### PR TITLE
More MemoryPool Unit Tests. Improve Re-broadcast back-off to an increasing linear formula.

### PR DESCRIPTION
**Pull request by jsolman**
I added more unit tests to MemoryPool. I have not found any problems with the code.

Also, this adds the one additional improvement I wanted to make so that the time for rebroadcasting re-verified transactions increases linearly as the pool size increases up until the pool size hits the capacity.

### CODE REVIEW

File: neo.UnitTests/UT_MemoryPool.cs [hunk 1]
```diff
@@ -89,24 +89,28 @@ public void TestSetup()
...
-            return tx;
+
+            return mockTx;
         }
 
         private Transaction CreateMockHighPriorityTransaction()
         {
-            var tx = CreateRandomHashInvocationTransaction();
+            var mockTx = CreateRandomHashInvocationMockTransaction();
+            mockTx.SetupGet(p => p.NetworkFee).Returns(Fixed8.FromDecimal(0.001m));
```
Comment (vncoelho) [inline neo.UnitTests/UT_MemoryPool.cs]: can fees be random with this limit here?
Maybe it will catch other cases.

File: neo/Ledger/MemoryPool.cs [hunk 2]
```diff
@@ -448,6 +449,9 @@ internal void UpdatePoolForBlockPersisted(Block block, Snapshot snapshot)
                 int blocksTillRebroadcast = unverifiedSortedTxPool == _sortedHighPrioTransactions
                     ? BlocksTillRebroadcastHighPriorityPoolTx : BlocksTillRebroadcastLowPriorityPoolTx;
 
+                if (Count > RebroadcastMultiplierThreshold)
+                    blocksTillRebroadcast = blocksTillRebroadcast * Count / RebroadcastMultiplierThreshold;
```
Comment (vncoelho) [inline neo/Ledger/MemoryPool.cs]: I will take another look to understand, capacity is quite big.

File: neo.UnitTests/UT_MemoryPool.cs (hunk 1 above)
Comment (vncoelho) [reply neo.UnitTests/UT_MemoryPool.cs]: I mean, random from, for example `[Fixed8.FromDecimal(0.001m), Fixed8.FromDecimal(0.1m)]`

Comment (vncoelho) [reply neo.UnitTests/UT_MemoryPool.cs]: They will still be high priority, but with different attached fees

File: neo/Ledger/MemoryPool.cs (hunk 2 above)
Comment (vncoelho) [reply neo/Ledger/MemoryPool.cs]: Capacity is not fixed, Jeff?
I did not exactly got this.

Currently, Capacity is 50000, right?
Thus `RebroadcastMultiplierThreshold = 5000` kind of fixed, right?

Then, for example.
If Count = 1000
blocksTillRebroadcast = 2 for high priority
blocksTillRebroadcast = 6 for low priority

If Count = 10000
blocksTillRebroadcast = 20 for high priority
blocksTillRebroadcast = 60 for low priority

Sounds reasonable.

Comment (jsolman) [reply neo/Ledger/MemoryPool.cs]: It just makes a linear progression as the pool fills up to in our current case the max Capacity of 50000 makes RebroadcastMultiplierThreshold=5000

For a capacity of 5000 or less:
* 30 blocks for rebroadcasting low priority
* 10 blocks for rebroadcasting high priority

From then on for every additional 5000 in the pool the time linearly increases by the same amount, so...

at 10000 in the pool:
* 60 blocks for rebroadcasting low priority 
* 20 blocks for rebroadcasting high priority

at 15000 in the pool:
* 90 blocks for low
* 30 blocks for high
...
at 50000 (full pool)
* 300 blocks for rebroadcasting low
* 100 blocks for rebroadcasting high

Also note that it doesnt jump at 5000 levels, for example @ 6000
* 36 blocks for rebroadcasting low priority
* 12 blocks for rebroadcasting high

Comment (vncoelho) [reply neo/Ledger/MemoryPool.cs]: Our comments were almost together...aehauehauea

Comment (jsolman) [reply neo/Ledger/MemoryPool.cs]: Yeah, you got it

Comment (vncoelho) [reply neo/Ledger/MemoryPool.cs]: I did not get this exception of `For a capacity of 5000 or less:`

Comment (vncoelho) [reply neo/Ledger/MemoryPool.cs]: Are you going to push it or it is already there?

Comment (jsolman) [reply neo/Ledger/MemoryPool.cs]: It’s already there

Comment (vncoelho) [reply neo/Ledger/MemoryPool.cs]: Anyway, sounds good to me, Jeff.
Congratulations. Nice idea.

Comment (vncoelho) [reply neo/Ledger/MemoryPool.cs]: `It’s already there`, I did not see...aehuaheuaea

**Pull request merged**